### PR TITLE
[MSCRT] Implement CONIO CRT Functions:  _getwch() and _getwche()

### PR DESCRIPTION
**Pull request by vicmarcal**
## Purpose

- This should fix Python MinGW
- FontForge
- Kicad 5.0.0

JIRA issues: [CORE-8028](https://jira.reactos.org/browse/CORE-8028), [CORE-11064](https://jira.reactos.org/browse/CORE-11064), [CORE-14865](https://jira.reactos.org/browse/CORE-14865)

## Proposed changes

- Implement `_getwch` and `_getwche`.
- Implementation comes from a patch of JIRA-8028 from CAv, updated by me. 
- As a reference, this is wine CONIO implementation counterparts and some other extras: https://source.winehq.org/git/wine.git/blob/HEAD:/dlls/msvcrt/console.c#l110

### CODE REVIEW

File: sdk/lib/crt/conio/getwche.c [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+ * COPYRIGHT:   See COPYING in the top level directory
+ * 		Copyright (C) 1994 DJ Delorie, see COPYING.DJ for details
+ * PROJECT:     ReactOS system libraries
+ * FILE:        lib/msvcrt/conio/getwche.c
+ * PURPOSE:     Reads a character from stdin
+ * PROGRAMER:   DJ Delorie
+				Ariadne
+				Lee Schroeder
+ * UPDATE HISTORY:
+ *              28/12/98: Created
```
Comment (HBelusca) [inline sdk/lib/crt/conio/getwche.c:11]: Please fix indentation in the whole file.

File: sdk/lib/crt/conio/getwche.c [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+ */
+
+#include <precomp.h>
+
+wint_t _getwche(void)
+{
+  if (char_avail)
+    /*
+     * We don't know, wether the ungot char was already echoed
```
Comment (HBelusca) [inline sdk/lib/crt/conio/getwche.c:20]:
```suggestion
     * We don't know, whether the ungot char was already echoed
```

File: sdk/lib/crt/conio/getwche.c [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+     * we assume yes (for example in cscanf, probably the only
+     * place where ungetch is ever called.
+     * There is no way to check for this really, because
+     * ungetch could have been called with a character that
+     * hasn't been got by a conio function.
+     * We don't echo again.
+     */
+    return(_getwch());
+  return (_putwch(_getwch()));
+ }
```
Comment (HBelusca) [inline sdk/lib/crt/conio/getwche.c:30]:
```suggestion
}

```

File: sdk/lib/crt/conio/getwch.c [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+            }
+        }
+        if (RestoreMode) {
+            SetConsoleMode(ConsoleHandle, ConsoleMode);
+        }
+    }
+    if (c == 10)
+        c = 13;
+    return c;
+}
```
Comment (HBelusca) [inline sdk/lib/crt/conio/getwch.c:123]:
```suggestion
}

```

**Review by HBelusca — COMMENTED** [review]
Points to review:
- code formatting+indentation
- what could be added there from Wine.

File: sdk/lib/crt/conio/conio.cmake [hunk 5]
```diff
@@ -4,6 +4,8 @@ list(APPEND CRT_CONIO_SOURCE
     conio/cputs.c
     conio/getch.c
     conio/getche.c
+    conio/getwche.c
+    conio/getwch.c
```
Comment (SergeGautherie) [inline sdk/lib/crt/conio/conio.cmake:8]:
```suggestion
    conio/getwch.c
    conio/getwche.c
```

**Pull request closed without merging**